src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -517,6 +517,24 @@ public function firstOrFail($columns = ['*'])
         throw (new ModelNotFoundException)->setModel(get_class($this->model));
     }
 
+    /**
+     * Execute the query and get the first result or throw an 403 forbidden exception.
+     *
+     * @param  array  $columns
+     * @param  string  $message
+     * @return \Illuminate\Database\Eloquent\Model|static
+     *
+     * @throws \Illuminate\Database\Eloquent\ModelForbiddenException
+     */
+    public function firstOrForbidden($columns = ['*'], $message = null)
+    {
+        if (! is_null($model = $this->first($columns))) {
+            return $model;
+        }
+
+        throw new ModelForbiddenException($message);
```

Comment:
Why do we need a new exception here? We can throw `AuthorizationException`.

Reply:
Actually I wanted to use AuthorizationException first. However, since it would give an error to someone trying to use the packages (such as illuminate/database, illuminate/auth) in the Laravel framework separately, I had to set them in the database.

Reply:
Good call!